src/writer/lsp_manager.py

```
@@ -0,0 +1,163 @@
+
+        Returns:
+            Available port number
+
+        Raises:
+            RuntimeError: If no available port is found
+        """
+        for port in range(start_port, end_port):
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+                try:
+                    s.bind((self.host, port))
+                    return port
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)